src/engagement_db_to_analysis/engagement_db_to_analysis.py

```diff
@@ -227,3 +229,7 @@ def generate_analysis_files(user, google_cloud_credentials_file_path, pipeline_c
         google_drive_upload.upload_all_files_in_dir(
             f"{output_dir}/automated-analysis", f"{drive_dir}/automated-analysis", recursive=True
         )
+
+    if pipeline_config.rapid_pro_target.sync_config.sync_advert_contacts:
+        sync_advert_contacts_to_rapidpro(participants_by_column, uuid_table, pipeline_config, rapid_pro,
+                             google_cloud_credentials_file_path, membership_group_dir_path, cache_path)
```

Comment:
It's interesting that you're doing this here vs. in the db -> rapid pro sync stage. 
I'm not too worried about it for now, especially as we'd only do this kind of sync within the context of a project. However, if it's necessary to do it here that suggests there's something wrong with the database itself which we should see if we can fix e.g. I can see that consent is going to get complicated in the database as the number of projects increases over time.

Reply:
I agree, I struggled with this a bit too. Choose to place this in the analysis stage so that in the future we have the flexibility to generate contacts based on the analysis logic e.g:
- based on location/ age_category imputations.
- project-based consent.
- etc
A future solution would be to write back imputed labels to engagement DB.